horace_core/sqw/PixelData/@pixobj_combine_info/pixobj_combine_info.m

```diff
@@ -55,32 +52,80 @@
             obj = obj.init(varargin{:});
         end
         function obj = init(obj,varargin)
-            [obj,remains] = init@MultipixBase(obj,varargin{:});
-            flds = {'run_label','filenum'};
+            flds = {'infiles','npix_list','run_label','filenum'};
             [obj, remains] = set_positional_and_key_val_arguments (obj, ...
-                flds, false, remains{:});
+                flds, false, varargin{:});
             if ~isempty(remains)
                 if numel(remains)==1
                     obj.filenum_ = remains{1};
                 else
                     error('HORACE:pixobj_combine_info:invalid_argument',[ ...
-                        'pixobj_combine_info accepts up to 5 input arguments.\n' ...
+                        'pixobj_combine_info accepts up to 4 input arguments.\n' ...
                         'got: %d arguments. Last have not been recognized: %s\n'], ...
                         numel(varargin),disp2str(remains))
                 end
             end
         end
         %------------------------------------------------------------------
+        function obj = init_pix_access(obj)
+            % initialize access to contributing pixels.
+
+            % as we normally read data and immediately dump them back, what
+            % is the point of converting them to double and back to sinlge?
+            % Keep precision.
+            for i=1:numel(obj.infiles_)
+                obj.infiles_{i}.keep_precision = true;
+            end
+        end
+
+        function [data,npix_chunk] = get_dataset_page(obj, ...
+                n_dataset,pix_pos,npix_idx)
+            % Return pixel data and pixel bin sub-distribution for the
+            % particular dataset ouf of multiple pixel datasets, stored
+            % within the class.
```

Comment:
I can't work out what this function does. Is this right: 

It takes a contiguous slice of pixels of a given size from the underlying `sqw` objects even if it covers multiple `sqw` objects


```suggestion
            % Return pixel data and pixel bin sub-distribution for the
            % particular dataset out of multiple pixel datasets, stored
            % within the class.
```

Reply:
its `pixobj_combine` object, which contains information about  pixels and images (pat of image -- `npix`) of all combining datasets. This method takes infor from one dataset. No `sqw` objects